### PR TITLE
Fix #14010: Refactor Velocity Literals to use _mph

### LINKED ISSUE #14010: Refactor Velocity Literals to use `_mph`
In our sister project I introduced the `_mph` literal for speeds https://github.com/OpenLoco/OpenLoco/pull/736
I'm almost definitely the underlying units are the same in OpenRCT2. Therefore we should bring in the same changes. First add the string literal code. Then find all of the velocity literals in the vehicle code.

Its quite easy to do just convert the literal to hex so `393216` = `0x60000` in hex then you drop the 4 `0`'s to get `0x6` which is the speed in `6.0_mph`

https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/ride/Vehicle.cpp#L5651
```
        if (velocity > -0x2C000)
            return OpenRCT2::Audio::SoundId::Null;
```
becomes
```
        if (velocity > -2.75_mph)
            return OpenRCT2::Audio::SoundId::Null;
```

You can use static_asserts to confirm that the values are identical `static_assert(2.75_mph == 0x2C000);`

### PR DESCRIPTION
**Pull request by tupaschoal**
I'm picking this up from #15517 because the source repository there has been deleted.

The conversation there was quite long, but seems like it had converged to what is currently here. Let me know if any other changes are needed.

PS: #14207 introduces the `_GBP` user defined literal, whereas this one adds `_mph`. It's probably best if we standardize like `_GBP` and `_MPH` or both lowercase, right? I can go either way.

### CODE REVIEW

**Comment by IntelOrca** [comment]
I think `_mph16` and `_mph` would be better. In other words, int32 should be the norm.

**Comment by tupaschoal** [comment]
> I think `_mph16` and `_mph` would be better. In other words, int32 should be the norm.

Done. I've also taken the opportunity to rename to uppercase, to conform with the other function to be introduced for money. In case we don't want, it's easy to drop it.

**Comment by Gymnasiast** [comment]
> It's probably best if we standardize like _GBP and _MPH or both lowercase, right? I can go either way.

I would keep them as GBP and mph. The reason for this is that it reflects their real-life usage. To me, it seems just a tad easier to read that way.

File: src/openrct2/ride/CableLift.cpp [hunk 1]
```diff
@@ -115,10 +115,10 @@ void Vehicle::CableLiftUpdate()
  */
 void Vehicle::CableLiftUpdateMovingToEndOfStation()
 {
-    if (velocity >= -439800)
+    if (velocity >= -6.710816_MPH)
```
Comment (Gymnasiast) [inline src/openrct2/ride/CableLift.cpp:118]: I’m reeeeeeally not sure if I like this. It feels finicky and prone to failure to me.

**Comment by tupaschoal** [comment]
> I would keep them as GBP and mph. The reason for this is that it reflects their real-life usage. To me, it seems just a tad easier to read that way.

Dropped the commit that made it uppercase

File: src/openrct2/ride/CableLift.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2/ride/CableLift.cpp]: And `kupt` is supposed to mean `kilo units per tick`?

I don't know, if it makes all numbers smaller, easier to read, seems like a good change, but then you also introduce a unit name that most people are unaware of?

Comment (duncanspumpkin) [reply src/openrct2/ride/CableLift.cpp]: There was a suggestions that the conversion is 2^15 and not 2^16 that locomotion uses. the way to tell would be comparisons between mph16 and mph32

Comment (tupaschoal) [reply src/openrct2/ride/CableLift.cpp]: Not sure I understand how this changes things

Comment (duncanspumpkin) [reply src/openrct2/ride/CableLift.cpp]: It would mean multiplying all mph32 values by 2 and using 32768 as the conversion instead of 65536

Comment (duncanspumpkin) [reply src/openrct2/ride/CableLift.cpp]: Ahh seems its none of the above: mph32 to mph16 is: `(vehicle->velocity * 9) >> 18;` i.e. 2.25 / 65536. So we need to do something like  `speedMph * 147456` so this user would become `2.9825_mph`.

**Comment by duncanspumpkin** [comment]
Change the factor to 29127 and i think this will be fine. It should produce sensible numbers then (or at least pretty sensible)

**Comment by tupaschoal** [comment]
> Change the factor to 29127 and I think this will be fine. It should produce sensible numbers then (or at least pretty sensible)

I did what you asked, but I'm not sure it's any better (converting everything to decimal then dividing by 29127). I think the calculation is off by one on the exponent here:

> (vehicle->velocity * 9) >> 18; i.e. 2.25 / 65536.

Isn't bitwise shifting `>> 18` equal to dividing by `2^17`? So this would become:
 9 / 131072 -> 4.5 / 65536 -> 2.25 / 32768 -> 1 / 14563,5

**Comment by tupaschoal** [comment]
@duncanspumpkin thoughts?

**Comment by duncanspumpkin** [comment]
> @duncanspumpkin thoughts?

I thought the changes look sensible now.

**Pull request closed without merging**